src/main/java/com/myMoneyTracker/app/filter/StatelessAuthenticationFilter.java

```diff
@@ -0,0 +1,67 @@
+package com.myMoneyTracker.app.filter;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import com.myMoneyTracker.app.service.SessionAuthentication;
+
+/**
+ * Class implementing {@link Filter} that will handle unauthorized access.
+ * 
+ * @author Florin
+ */
+public class StatelessAuthenticationFilter implements Filter {
+    
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+    
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+        
+        if (isAllowedURL(httpRequest.getRequestURI())) {
+            chain.doFilter(request, response);
+        } else {
+            String headerToken = httpRequest.getHeader("mmtlt");
+            SessionAuthentication sessionAuthentication = (SessionAuthentication) SecurityContextHolder.getContext().getAuthentication();
+            String sessionToken = sessionAuthentication == null ? null : sessionAuthentication.getSessionToken();
+            if (headerToken != null && headerToken.equals(sessionToken)) {
+                chain.doFilter(request, response);
+            } else {
+                httpResponse.setStatus(401);
+                response.getOutputStream().print("Unauthorized access!");
+            }
+        }
+    }
+    
+    private boolean isAllowedURL(String url) {
+    
+        boolean isAllowed = false;
+        if (url != null) {
+            if (url.contains("?")) {
```

Comment:
To clean up a very long path. We need to call contains method many times, and is necessary to 
have a shorter path before performing contains for each allowed string.
Long path example:
'https://www.google.ro/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=whats+happening+if+you+text+a+very+long+path+into+the+URL'